telephony/android_modem.c

```diff
@@ -3351,12 +3364,25 @@ amodem_addTimeUpdate( AModem  modem )
     * and deal with this case (since it normally relied on the operator's country code
     * which is hard to simulate on a general-purpose computer
     */
-    amodem_add_line( modem, "%%CTZV: %02d/%02d/%02d:%02d:%02d:%02d%c%d:%d:%s\r\n",
-             (utc.tm_year + 1900) % 100, utc.tm_mon + 1, utc.tm_mday,
-             utc.tm_hour, utc.tm_min, utc.tm_sec,
-             (tzdiff >= 0) ? '+' : '-', (tzdiff >= 0 ? tzdiff : -tzdiff) / 15,
-             (local.tm_isdst > 0),
-             tzname );
+    amodem_add_cvtz_line(modem, &utc, &local, tzdiff, tzname);
+}
+
+/**
+ * Update the NITZ timezone to tzdiff from UTC in number of quarter-hours.
+ */
+void
+amodem_set_timezone( AModem modem, int tzdiff )
+{
+    time_t       utc_now = time(NULL);
+    time_t       local_now = utc_now + tzdiff * 15 * 60;
+    struct tm    utc, local;
+
+    utc   = *gmtime( &utc_now );
+    local = *gmtime( &local_now );
+
+    amodem_begin_line( modem );
+    amodem_add_cvtz_line(modem, &utc, &local, tzdiff, "Unknown/Unknown");
```

Comment:
In [line#3348](https://github.com/mozilla-b2g/platform_external_qemu/pull/162/files?diff=split#diff-58bd35ac9e1da86dffbc1dd35cd1ccc0L3335), we replace every '/' in the timezone name by '!'. Do we need to do the same replacement here, although neither rild nor gecko reads this timezone name actually?